Add vend support to go-modules

c00w opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/1/nixpkgs/manual.html#chap-reviewing-contributions
-->

###### Motivation for this change
Currently you have to manually patch vendor directories that distribute c code via the go package system. By adding the useVend option, this gets much simpler.

Fixes #89128

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

c00w committed:
vend: init at unstable-2020-06-04

This is used to provide alternate vendoring for go.

c00w committed:
go-modules: Add useVend flag to go-modules

c00w committed:
go-ethereum: use vend to simplify package

c00w committed:
hugo: use vend to simplify package

c00w committed:
blockbook: use vend to simplify package

c00w committed:
saml2aws: use vend to simplify package

zowoq commented on pkgs/development/tools/vend/default.nix:
```diff
@@ -0,0 +1,21 @@
+{ stdenv, buildGoModule, fetchFromGitHub }:
+
+buildGoModule {
```
I'm kind of surprised that this builds without `vendorSha256 = null` being set?

c00w replied:
Default is null for vendorSha256 :) So if your package is fully vendored, you don't need to do anything, just set src and it will build :D

zowoq replied:
Why are we setting `null` in packages then?

zowoq replied:
Opened a PR: https://github.com/NixOS/nixpkgs/pull/89454

c00w commented:
Had to open
https://github.com/nomad-software/vend/pull/9 so that vend wouldn't run go mod tidy which isn't deterministic.

Mic92 commented on pkgs/development/go-modules/generic/default.nix:
```diff
@@ -20,6 +20,8 @@
 # Whether to delete the vendor folder supplied with the source.
 , deleteVendor ? false
+# Whether to use the vend tool to regenerate the vendor directory.
+, useVend ? false
```
Would create any harm to run this by default beside potential bugs in the tool?

Mic92 replied:
mhm. It might break our checksums again... For that reason we might want to test it.

Mic92 replied:
`useVend` is not very descriptive. Could be rename it to something like vendor C code?

c00w replied:
A few arguments against running this by default.
1) This is a lot more cpu expensive, it's longer to run and downloads everything twice.
2) I don't expect the upstream support will be very good compared to go. If go updates to a new version, I don't want that version updated blocked for everything by vend support.
3) This is a super weird thing to do. Almost no one else uses this program, and it's super fringe. I don't want nix users encountering too many differences from the normal go development tooling unless we actually need it. I know if I encountered bugs caused by this program when packaging a bog standard well behaved no c code go program that works with the standard tooling, I'd be mightily upset.

c00w replied:
useVend -> runVend. The key is to make sure that reading the option conveys we're running a pgroam to a user. vendorFull makes this sound like some supported thing, rather than a weird program that helps in a couple of corner cases.

kalbasit reviewed (approved):
Would be great if we can run this by default, assuming it does not affect the current hashes as @Mic92 mentioned. Thanks for working on this!

Mic92 commented:
What I am a bit worried about is that new versions of goVend might include new files. For that reason might not want to include it for all packages.

c00w commented:
Vend rejected my patch, so we're going to have to maintain a semi permanent fork of it anyway :( Everything else is updated.

Mic92 commented on pkgs/development/go-modules/generic/default.nix:
```diff
@@ -88,6 +92,11 @@ let
+
+      if [ ${vendFlag} == "true" ]; then
+        echo running vend to rewrite vendor folder
+        ${vend}/bin/vend -commands=download,vendor
```
I actually prefer the optionalString variant because it does not require every go build to have download `vend`.

c00w replied:
I think I did this, let me know if my fix isn't what you intended.

Mic92 commented on pkgs/development/go-modules/generic/default.nix:
```diff
@@ -20,6 +20,8 @@
 # Whether to delete the vendor folder supplied with the source.
 , deleteVendor ? false
+# Whether to run the vend tool to regenerate the vendor directory.
+, runVend ? false
```
Can you also add this to documentation in `languages-frameworks/go.xml`?

c00w replied:
Done.

Mic92 commented on pkgs/development/go-modules/generic/default.nix:
```diff
@@ -20,6 +20,8 @@
 , vendorSha256 ? null
 # Whether to delete the vendor folder supplied with the source.
 , deleteVendor ? false
+# Whether to run the vend tool to regenerate the vendor directory.
```
```suggestion
# Whether to run the vend tool to regenerate the vendor directory. This is useful if libraries contain C files.
```

c00w replied:
Added

Mic92 commented on pkgs/development/go-modules/generic/default.nix:
```diff
@@ -20,6 +20,8 @@
 , vendorSha256 ? null
 # Whether to delete the vendor folder supplied with the source.
 , deleteVendor ? false
+# Whether to run the vend tool to regenerate the vendor directory.  # This is useful if any dependency contain C files.
```
```suggestion
# Whether to run the vend tool to regenerate the vendor directory.  
# This is useful if any dependency contain C files.
```

c00w replied:
Fixed :)

c00w commented:
@Mic92  Fixed formatting :)

marsam commented on pkgs/development/go-modules/generic/default.nix:
```diff
@@ -20,6 +20,9 @@
 , deleteVendor ? false
+# Whether to run the vend tool to regenerate the vendor directory.
+# This is useful if any dependency contain C files.
+, runVend ? false
```
what do you think of using `useVend`?

Mic92 replied:
We had this before https://github.com/NixOS/nixpkgs/pull/89453#discussion_r435156648 I thought about something more descriptive without having a better idea.

c00w replied:
Sounds like this is staying runVend without a 3rd alternative.

marsam commented on pkgs/development/go-modules/generic/default.nix:
```diff
@@ -88,6 +93,11 @@ let
       fi
       go mod vendor
+
+      if [ ${vendCommand} != "false" ]; then
```
IIUC we can use:
```diff
diff --git i/pkgs/development/go-modules/generic/default.nix w/pkgs/development/go-modules/generic/default.nix
index 7bfeb40dd7e..41010d92ef1 100644
--- i/pkgs/development/go-modules/generic/default.nix
+++ w/pkgs/development/go-modules/generic/default.nix
@@ -92,11 +92,12 @@ let
         echo "vendor folder exists, please set 'vendorSha256=null;' or 'deleteVendor=true;' in your expression"
         exit 10
       fi
-      go mod vendor
 
       if [ ${vendCommand} != "false" ]; then
         echo running vend to rewrite vendor folder
         ${vendCommand}
+      else
+        go mod vendor
       fi
       mkdir -p vendor
 ```

c00w replied:
Flipped.

marsam commented on pkgs/development/tools/vend/default.nix:
```diff
@@ -0,0 +1,22 @@
+
+  meta = with stdenv.lib; {
+    homepage = "https://github.com/c00w/vend";
+    description = "A utility which vendors go code including c dependencies.";
```
nitpick, `description` shouldn't contain periods
```suggestion
    description = "A utility which vendors go code including c dependencies";
```

c00w replied:
Done

marsam commented on pkgs/development/tools/vend/default.nix:
```diff
@@ -0,0 +1,22 @@
+  pname = "vend";
+  version = "unstable-2020-06-04";
+
+  # A permanent fork from master is maintained to avoid non deterministic go tidy
```
can we just fetch https://github.com/c00w/vend/commit/49899829b1e4bcd22ee3ba2d12bc76863196cb03 as a patch?

Mic92 replied:
I don't mind if we had our own fork so we can make sure it does not break anything in nixpkgs.

c00w replied:
*shrug* flipped to patch.

c00w commented:
@Mic92  - I think nothing else is outstanding here.

marsam commented:
IMHO `useVend` would make a bit more sense, but I don't have a strong opinion about it. Its description could be:

+ `useVend`: Whether to use [vend](https://github.com/nomad-software/vend) instead of `go mod vendor` to vendorize the go module dependencies

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-already-reviewed/2617/174

zowoq commented:
Removed trailing whitespace and formatted commit messages, I'll merge after builds pass on darwin.

@ofborg build blockbook go-ethereum gobetween hugo saml2aws

zowoq commented:
@c00w Can we use this for `aerc` and `mautrix-whatsapp`?

mdlayher commented on pkgs/development/tools/vend/default.nix:
```diff
@@ -0,0 +1,26 @@
+  vendorSha256 = null;
+
+  meta = with stdenv.lib; {
+    homepage = "https://github.com/c00w/vend";
```
Nit: out of date link now that you're using upstream plus a patch, rather than a fork, right?

zowoq replied:
Not sure that we want to point to upstream as we're applying a patch that they have refused.

mdlayher replied:
Ack, consider this resolved then.

mdlayher reviewed (approved):
LGTM, seems like a nice simplification!

c00w commented:
@zowoq aerc and mautrix-whatsapp don't build with it enabled. Not sure why, haven't put time in to figure it out.

zowoq commented:
> aerc and mautrix-whatsapp don't build with it enabled.

I'm not sure if we should merge this then as it doesn't work for two of the seven packages that have this workaround.

c00w commented:
I'm confused. The whole point of this PR is to add some tools and make package maintainers lives easier. Just because it doesn't fix literally every usage of overrideModAttrs doesn't mean we should reject it. It still dramatically simplifies a bunch of other packages and provides a nice base to delve into aerc and mautrix (which have some weird bug related to native dependencies at build time)...

Honestly given how much feedback was received on this, we probably should have merged this much earlier than 21 days after the first PR :)

zowoq commented:
AFAIK implementing `goDeps` / `vgo2nix` for `buildGoModule` would also resolve these problems and from the discussions in the other issues it seems like we're going to add it anyway.

I'm not sure if we should add an option that doesn't seem to work consistently for so few packages (5 of 7 out of ~230).

c00w commented:
When dep support lands for go modules, we can revert this code, until then this is solving a real problem for packagers, that's basically undiscoverable unless you really understand the details (I think @Mic92 and myself did all of the current fixes).

c00w commented:
@doronbehar  @betawaffle any chance you can lend some support here? You both seemed to care about this from the issue.

betawaffle commented:
I'm in support of this getting merged. It's very disturbing that purely module-based builds produce a different result than vendor'd builds. I think it's a failing of the go tooling, but more importantly, the deprecation of the previous `buildGoModule` means that vendoring always happens! So this seems like required functionality for when the change to vendoring becomes required.

zowoq commented:
> When dep support lands for go modules, we can revert this code, until then this is solving a real problem for packagers

Even if this is a temporary solution shouldn't we at least know why it doesn't work for some packages before we merge it?

c00w commented:
@zowoq We don't turn it on for those packages. Given we're not merging this for 5/7 packages, I'm really hesitant to invest effort in fixing the last two packages, since I feel like it's going to be wasted and not merged.

zowoq commented:
> We don't turn it on for those packages. 

I don't understand what this means.

> Given we're not merging this for 5/7 packages, I'm really hesitant to invest effort in fixing the last two packages, since I feel like it's going to be wasted and not merged.

If the packages are fixed I don't see an issue with merging this.

c00w commented:
>> We don't turn it on for those packages.
> I don't understand what this means.
This is an option that has to be enabled on a per package basis to have an effect. This PR turns it on for 5 packages, simplifying the complexity of their packaging. The other 2 packages this doesn't magically fix the build for, still build since we're not turning on this option for them.

>> Given we're not merging this for 5/7 packages, I'm really hesitant to invest effort in fixing the last two packages, since I feel like it's going to be wasted and not merged.
> If the packages are fixed I don't see an issue with merging this.

Look - this is really frustrating. I'm going out of my way to clean up some complexity in nixpkgs. You're blocking this because I didn't fix every possible bit of complexity. Merging this doesn't introduce more complexity and does a decent job of cleaning up a bunch of code. In general, I prefer to do stuff incrementally, and I'm not going to commit more and more time to something that I can't merge a basic step for.

Also just to be clear, the 2 packages that this won't work for, don't have this enabled yet, so they won't break... I don't understand why the last 2 packages are a blocker for merging.

Also given you said
> Removed trailing whitespace and formatted commit messages, I'll merge after builds pass on darwin.
and then didn't merge, I'm really not excited about investing more effort here since I expect another reason to not merge will come up and it'll be wasted.

zowoq commented:
I'll clarify my position:

I was going to merge this and noticed that two packages weren't included without being mentioned so I asked about them.

I think we both agree that `runVend` should work for these packages but it doesn't.

If we can't get those packages to work with `runVend` we should find out why so we have some assurance that it isn't going stop working for the packages it does currently work with.

c00w committed:
aerc: update to use runvend

c00w committed:
mautrix-whatsapp: update to use runVend

c00w commented:
> I think we both agree that runVend should work for these packages but it doesn't.

Nope - perfect is the enemy of the good, we should just merge even without those packages working. I've said this repeatedly + stand by it.

> If we can't get those packages to work with runVend we should find out why so we have some assurance that it isn't going stop working for the packages it does currently work with.

I think telling someone who is making an improvement that their work wasn't perfect so you're rejecting it is a pretty  rude attititude. You have a massive power differential over someone without commit powers. Not everyone has infinite time to commit to open source, let alone run down something minor that you're worried about. In this case if it broke in the future, falling back to the old format (which is in the git history) would be pretty acceptable. It also shows a pretty stark lack of empathy for packagers who have been asking for this for months :(. I personally am also feeling massively frustrated, to the point where a) I'm working on tooling to maintain a hard fork of nixos, and b) seriously considered not upstreaming improvements to nix for fear that you would review them.

alyssais commented:
Hi! First off, I don’t know much about the go ecosystem, so I don’t have anything to say either way on whether this change should be merged or not. I am, however, worried about the direction this discussion is going in.

@c00w, thank you very much for your pull request. We all appreciate the time you’ve taken to help make Nixpkgs better.

If I understand correctly, @zowoq expressed intention to merge this PR once certain things happened, and then once those things happened, found another thing that concerned them and brought that up rather than merge the PR. Because of this you’ve been feeling worried that even if you go through the work to figure out why this change doesn’t work the remaining two packages, it still won’t be merged.

I’m afraid I can’t really say anything to assure you it will be, because this sort of thing does happen, and there’s not very much any of us can do about it. Committers have to use their best judgement when deciding whether to merge things, and sometimes something just comes up that wasn’t anticipated before when somebody is giving something what was supposed to be a final look over. We can express our intentions to accept a contribution at a certain point, but can’t guarantee it. Even if something we just noticed has been the case since the start of the PR, it’s sometimes the case that, as humans, we didn’t notice it until it was too late.

You might be thinking “what harm does merging this do, even if it’s incomplete?” One reason it might not be worth doing is that because it wouldn’t replace the existing way of doing things, merging it would increase the complexity of Go packaging in general. This could make it harder to understand existing go packages (one more thing to learn), harder to add new packages (should I use this thing or not? Some packages do it this way, and some another. Why is that?), etc. Another reason is that often, once a feature has been merged, authors don’t come back to resolve problems with it. (I’ve certainly been guilty of this before.) This means that features have to be evaluated based on the current implementation, rather than taking into account potential future improvements. It can also mean that reviewers push back a bit more against things that don’t seem to be complete than they might otherwise need to, because it’s quite likely to be the only time somebody is ever going to be willing to work on the feature. Sometimes, this sort of pushback is justified, and sometimes it isn’t.

You might feel that the advantages outweigh the disadvantages here, but it’s not unreasonable for somebody else to balance those needs differently. In general, I don’t think that expecting a feature to work consistently before accepting it is uncommon in open source software projects — sometimes the feature is compelling enough that it’s worth accepting anyway, and sometimes it isn’t. That is a judgement call that people have to make.

Addressing some specific points you made:

> You have a massive power differential over someone without commit powers[…] It also shows a pretty stark lack of empathy for packagers who have been asking for this for months

This is one of the hard things about being a committer to large project, and is something that committers absolutely should keep in mind at all times. It’s also something that, in my opinion, we try to take into account as a project — for example, we try to reduce this power differential is by making many contributors committers, and trying to do so relatively quickly in response to repeated good contributions to Nixpkgs.

Is there something specific you believe has happened that was an abuse of this power? I don’t think asking for changes based on a newly-identified issue is one, based on what I’ve seen here. Similarly, I don’t think it’s unempathetic.

> I personally am also feeling massively frustrated, to the point where a) I'm working on tooling to maintain a hard fork of nixos, and b) seriously considered not upstreaming improvements to nix for fear that you would review them.

Talking about being frustrated is fine, but I don’t think it’s okay to threaten a fork while trying to get a change accepted. You’re very welcome to fork Nixpkgs, but threatening to isn’t going to help in any way. In fact I think it’s probably going to make people reading this sympathetic to your situation.

***

Sorry for the wall of text here. I want you to know that while I don’t think anybody has really done anything wrong here, I understand how frustrating this experience has been to you, and that that frustration is legitimate.

I do wonder if there’s anything we could do to move this along even if there are outstanding issues. For example, I haven’t seen any issues raised in regards to the addition of the “vend” package, so I’d be happy to cherry-pick that commit if you’d like. Then there’s one less thing to worry about here.

It might also be a good idea to just take a step back or take a break here, and let yourself calm down a bit. I have to do this myself sometimes. If you decide this has gone on long enough and you don’t want to deal with it any more, it might be better to just walk away.

Mic92 commented:
@c00w It looks like you managed to fix everything, which was the objection in the first place. @zowoq I think we can merge this?

c00w commented:
@alyssais

I appreciate you stepping in. I'm still very concerned since it appears that we're just sweeping away the issues (When should we merge a PR? Should we allow incremental fixes vs permanent ones? What does appropriate behavior look like? How much does reported user pain matter vs incremental work? Should we have reviewers who disagree with code someone is writing be allowed to pick up their reviews? How do we handle disagreements on design decisions? Do reviewers have enough support here? I may very well be overwhelming people with my questions + driving someone else out of the community).

These aren't fun discussions (I'm certainly not having any fun and I expect no one else is either), but they're important ones.

Also feel free to tell me I'm just going off the deep end and should shut up (I'm sort of assuming that's not what you're trying to do, but if you are feel free to be blunt - I'll shut up and go away).

c00w commented:
@Mic92 The issue is that there is still some hand patching for go modules. So my literal interpretation is that this won't be approved for merge unless we fix all of it (which is likely impossible).

Mic92 commented:
> @Mic92 The issue is that there is still some hand patching for go modules. So my literal interpretation is that this won't be approved for merge unless we fix all of it (which is likely impossible).

How many packages are left?

c00w commented:
@Mic92 - Looks like it's tinygo.
```
grep overrideModAttrs pkgs -r
pkgs/development/go-modules/generic/old.nix:, overrideModAttrs ? (_oldAttrs : {})
pkgs/development/go-modules/generic/old.nix:  args = removeAttrs args' [ "overrideModAttrs" "modSha256" "disabled" ];
pkgs/development/go-modules/generic/old.nix:  ) // overrideModAttrs modArgs);
pkgs/development/go-modules/generic/default.nix:, overrideModAttrs ? (_oldAttrs : {})
pkgs/development/go-modules/generic/default.nix:  args = removeAttrs args' [ "overrideModAttrs" "vendorSha256" "disabled" ];
pkgs/development/go-modules/generic/default.nix:  ) // overrideModAttrs modArgs) else "";
pkgs/development/compilers/tinygo/default.nix:  overrideModAttrs = (_: {
```

zowoq commented:
`tinygo` is a more complicated package (compiler vs leaf cli tools, etc) and it has a different issue (with a different fix) than the packages covered by this PR. I wasn't expecting it to be fixed as part of this.

The two packages I did ask about are building with `runVend` now, so as I said previously I'm fine with merging this now.

`hugo` looks like it's failing with an incorrect `vendorSha256`?

Mic92 commented:
Maybe it was updated since than.

zowoq commented:
Looks like the commit the `vend` package is using (https://github.com/nomad-software/vend/pull/10/commits/1c66266a7a9ab98897683bbc4f0a6dc26328a44a from https://github.com/nomad-software/vend/pull/10) fixes `aerc` and `mautrix-whatsapp` but breaks `hugo`.

Mic92 commented:
Result of `nixpkgs-review pr 89453` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages failed to build:</summary>
<br>- go-ethereum
<br>- hugo
<br>- mautrix-whatsapp
</details>
<details>
  <summary>5 packages built:</summary>
<br>- aerc
<br>- blockbook
<br>- gobetween
<br>- saml2aws
<br>- vend
</details>

Mic92 commented:
@GrahamcOfBorg build go-ethereum hugo mautrix-whatsapp

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/using-go-mod-edit-in-packages/8305/2

alyssais commented:
> I appreciate you stepping in. I'm still very concerned since it
> appears that we're just sweeping away the issues (When should we merge
> a PR? Should we allow incremental fixes vs permanent ones? What does
> appropriate behavior look like? How much does reported user pain
> matter vs incremental work? Should we have reviewers who disagree with
> code someone is writing be allowed to pick up their reviews? How do we
> handle disagreements on design decisions? Do reviewers have enough
> support here? I may very well be overwhelming people with my questions
> + driving someone else out of the community).

I don't want to distract from the code in this thread, but discussions
in this area are absolutely important -- if you'd like to talk more to
me about it (no pressure!) I'm qyliss on IRC. :)

> These aren't fun discussions (I'm certainly not having any fun and I
> expect no one else is either), but they're important ones.

Totally agree.

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/how-many-people-are-paid-to-work-on-nix-nixpkgs/8307/9

zowoq commented:
`gobetween` can be dropped from this PR. https://github.com/NixOS/nixpkgs/pull/93972

c00w commented:
gobetween is dropped - working on figuring out hugo (looks like a rebase broke it).

c00w commented:
And hugo is fixed - should be g2g

zowoq commented:
`go-ethereum` builds but the `vendorSha256` needs to be updated.

```diff
diff --git a/pkgs/applications/blockchains/go-ethereum.nix b/pkgs/applications/blockchains/go-ethereum.nix
index e2b308d88e2..0edeccf4003 100644
--- a/pkgs/applications/blockchains/go-ethereum.nix
+++ b/pkgs/applications/blockchains/go-ethereum.nix
@@ -12,7 +12,7 @@ buildGoModule rec {
   };
 
   runVend = true;
-  vendorSha256 = "0r08fayxpccjyr29g38wc0zvpqgma9inyvdwcama1fyiq0n8n39d";
+  vendorSha256 = "1744df059bjksvih4653nnvb4kb1xvzdhypd0nnz36m1wrihqssv";
 
   subPackages = [
     "cmd/abidump"
``` 

Can you push the changes in https://github.com/nomad-software/vend/commit/24fdebfdb2c3cc0516321a9cf33a3fd81c209c04 to https://github.com/nomad-software/vend/pull/10 please?

Upstream seems to be responsive so I'll to give them a day or two to comment on the new changes.

c00w commented:
Updated go-ethereum.

Re: vend - somehow I updated the pr verbally but didn't push. I'd strongly prefer we don't block on vend upstream, since they're still not convinced that the PR is needed + they've literally rejected every PR I sent them related to nix functionality. If they do merge it I can flip the repos back, but it's not great odds.

zowoq commented:
> Re: vend - somehow I updated the pr verbally but didn't push. I'd strongly prefer we don't block on vend upstream, since they're still not convinced that the PR is needed + they've literally rejected every PR I sent them related to nix functionality. If they do merge it I can flip the repos back, but it's not great odds.

If they are willing to accept the functionality but request changes we may need to update packages `vendorSha256` again?

Mic92 commented:
> > Re: vend - somehow I updated the pr verbally but didn't push. I'd strongly prefer we don't block on vend upstream, since they're still not convinced that the PR is needed + they've literally rejected every PR I sent them related to nix functionality. If they do merge it I can flip the repos back, but it's not great odds.
> 
> If they are willing to accept the functionality but request changes we may need to update packages `vendorSha256` again?

I wonder if we just should maintain our own fork anyway. The package is quite simple and we cannot have upstream breaking our hashes if @r-ryantm makes a PR and somebody merges it without knowing the knowing the consequences.

c00w commented:
@zowoq @Mic92 In what proves my prothetic abilities - they just rejected the PR :P

I can move this to nix-community (although I'd prefer if I could get admin permissions there on the vend repo).

zowoq commented:
@Mic92 Is this okay to merge or do you want to do the switch to nix-community in this PR?

Mic92 commented:
> @zowoq @Mic92 In what proves my prothetic abilities - they just rejected the PR :P
> 
> I can move this to nix-community (although I'd prefer if I could get admin permissions there on the vend repo).

I sent you and invite. Once you moved the project their, you will be the maintainer and marked as such.
However it also allows to easily move over maintenance in case you are lacking time.

Mic92 commented:
> @Mic92 Is this okay to merge or do you want to do the switch to nix-community in this PR?

Let's just move before. The process is just a few clicks away.

zowoq commented:
Checking darwin:

@ofborg build aerc blockbook go-ethereum hugo mautrix-whatsapp saml2aws
